server/services/source-map.js

```diff
@@ -7,20 +7,49 @@ const libSourceMaps = istanbulAPI.libSourceMaps;
 
 const sourceMap = libSourceMaps.createSourceMapStore({verbose: Conf.IS_COVERAGE_ACTIVE});
 const meteor_dir = Conf.COVERAGE_APP_FOLDER;
-const regexAlterationSourceMapPath = new RegExp(/(packages\/)([a-zA-Z-]*)[_:]([a-zA-Z-_]*)(.*)/);
+// Next RegExps must consider following paths:
+//  - meteor://💻app/packages/pkgAuthor[_:]pkgName/...
+//  - meteor://💻app/packages/local-test:pkgAuthor:pkgName/...
+const regexAlterationSourceMapPath = new RegExp(/(packages\/)(?:local-test[_:])?([a-zA-Z-]*)[_:]([a-zA-Z-_]*)(.*)/);
+const regexPUT = new RegExp(/packages\/local-test:([a-zA-Z-_]*):([a-zA-Z-_]*)\/.*/);
 
 // Alter inside the source map the path of each sources
 alterSourceMapPaths = function (map) {
   var match;
+
+  if (!Meteor.PUT && Meteor.isPackageTest && meteor_dir === `${process.env.PWD}/`) {
```

Comment:
I've created that new `Meteor` property because the only time I can get the package name/author is when `registerSourceMap()` is called with the test file (for example, `client.tests.js`) in `map.sources` array.

That first time I've seen it does not include the other files (app/packages; for example). The next call to `registerSourceMap()` includes all the involved files (app/packages), but not the test file, so I cannot get the package name/author that second time. I need some kind of persistence between the calls, and the `Meteor` object is the first thing I have thought of.

I make the comparsion `meteor_dir === ${process.env.PWD}/` because I want to check if the `COVERAGE_APP_FOLDER` is the same as meteor app root. If both are the same (with the end backslash exception), then user is testing the package from its folder, and not outside.

User could do `meteor test-packages --driver-package practicalmeteor:mocha packages/my-package`. In that case, the paths resolve successfully without this PR. That's why I make that comparison.

I don't know if there is a better way to do this, but it's the only that has come to my mind.

Any suggestion/improvement will be welcome :blush: